sdk/appconfig/azappconfig/client.go

```diff
@@ -140,11 +140,19 @@ func fromGeneratedAdd(g generated.AzureAppConfigurationClientPutKeyValueResponse
 
 // AddSettingOptions contains the optional parameters for the AddSetting method.
 type AddSettingOptions struct {
-	// placeholder for future options
+	// Configuration setting label.
+	Label *string
 }
 
 // AddSetting creates a configuration setting only if the setting does not already exist in the configuration store.
-func (c *Client) AddSetting(ctx context.Context, setting Setting, options *AddSettingOptions) (AddSettingResponse, error) {
+func (c *Client) AddSetting(ctx context.Context, key string, value *string, options *AddSettingOptions) (AddSettingResponse, error) {
```

Comment:
Is specifying a `nil` value for `value` the way to "erase" a setting?  If so it would be good to document that, else it gives the appearance that `value` is optional.

Reply:
Value can be `nil`, or can be `""`. AppConfig does distinguish between them, i.e. if you set `nil`, you read `nil` back, the same for `""`. In the Azure Portal, you can't tell the difference.